docs/blog/2024-07-25-nx-19-5-update.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Nx 19.5 Update!!
+slug: 'nx-19-5-update'
+authors: ['Zack DeRose']
+cover_image: '/blog/images/2024-07-25/nx-19-5-thumbnail.png'
+tags: [nx, release]
+---
+
+## NEW: Pattern Support for `targetDefaults`
+
+We've added Pattern Support for `targetDefaults` in your `nx.json` file.
+
+Generally speaking, `targetDefaults` are a way for you to specify default configurations for your workspace. For example, if for your workspace, your `test` tasks always depend on your `build` task to have run before the `test`s can run, you'd add this as a `targetDefault`:
+
+```json
+{
+  "targetDefaults": {
+    "test": {
+      "dependsOn": ["build"]
+    }
+  }
+}
+```
+
+By adding pattern matching to `targetDefault`s, we can support a wider range of target names, which fits in really well with recent enhancements from Project Crystal, like task atomization. For example, the `@nx/playwright` plugin will automatically add in a task with the name `e2e-ci--${testFileName}` for every playwright spec file in your project. With this new pattern support in `targetDefaults`, we can define default behavior for all these tasks.
+
+The following will set all such tasks to depend on their `build` task:
+
+```json
+{
+  "targetDefaults": {
+    "e2e-ci--**/*": {
+      "dependsOn": ["build"]
+    }
+  }
+}
+```
+
+## NEW: Individual Targets Can Now Opt Out of Parallelism
+
+The ability to "Atomize" larger tasks by automatically breaking them up into smaller pieces is a really cool enhancement that is enabled by Nx Project Crystal. Especially when combined with the ability to distribute tasks across many machines by using Distributed Task Execution from Nx Agents.
+
+{% youtube
+src="https://youtu.be/_FSHQIwITic"
+title="Nx Agents"
+width="100%" /%}
+
+This approach can break down a massive and long-running end-to-end test suite into many smaller tasks. Then you can divide the time it takes to run the entire suite by the number of machines provided via Nx Agents.
+
+However, previously when using this approach, there were some limitations regarding parallelism. Most end-to-end tests cannot run on the same machine at the same time due to shared ports, conflicting database states, etc.
+
+The solution in the past was to add a `--parallel=1` option, which was a bit clunky. For example:
+
+```terminal
+> nx run-many --targets=lint,test,e2e,build --parallel=1
```

Comment:
I think `shell` might be the correct one here, as this one doesn't display in the usual terminal window style.

Also the solution in the past in the CI config was to have two separate lines, like

```
> nx affected --targets=lint,test,build
> nx affected --targets=e2e-ci --parallel=1
```

Reply:
I'd probably even include a small excerpt of the CI config that we generate, with "..." out the non-relevant sections at the top of it. I'm thinking about something like what is being done here:

https://canary.nx.dev/ci/intro/tutorials/github-actions#parallelize-tasks-across-multiple-machines-using-nx-agents

(there's a missing `e2e-ci` in that example though right now which Jack is going to fix)